src/parser/jobs/gnb/modules/AmmoCombo.tsx

```diff
@@ -0,0 +1,262 @@
+
+// Expected within a standard gnashing.
+const EXPECTED_USES = {
+	JUGULAR_RIP: 1,
+	SAVAGE_CLAW: 1,
+	ABDOMEN_TEAR: 1,
+	WICKED_TALON: 1,
+	EYE_GOUGE: 1,
+}
+
+const GNASHING_TIME = 7500 // time in ms, this is the "fastest" time a combo can happen, but can take longer TODO: Dropped Combos?
+
+class GnashingComboState {
+	start: number
+	end?: number
+	rotation: CastEvent[] = []
+	// isRushing: boolean = false //I've yet to figure out how I wanted to handle end of fights rushes
+	isProper: boolean = false
+
+	// the trackers
+	numGnashing: number = 0
+	numJugular: number = 0
+	numSavage: number = 0
+	numAbdomen: number = 0
+	numWicked: number = 0
+	numEye: number = 0
+	numTotal: number = 0 // This is the filter, sub 6 = possible problem
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
+export default class AmmoCombo extends Module {
+	static handle = 'Gnashing Fang Combo issues'
+
+	@dependency private checklist!: Checklist
+	@dependency private timeline!: Timeline
+
+	private buffs = 0
+	private actions = 0
+
+	private gnashingComboWindows: GnashingComboState[] = [] // Store all the gnashing fang combos to be output to the rotationTable
+
+	private get lastGnashingCombo(): GnashingComboState | undefined {
+		return _.last(this.gnashingComboWindows)
+	}
+
+	protected init() { // TODO: Strip away the other hooks? I could roll them into the actual cast I have for tracking the combo, but won't be as nice looking IMO
+		this.addHook('cast',
+			{
+				by: 'player',
+				abilityId: RELEVANT_ACTIONS,
+			},
+			() => this.actions++)
+
+		this.addHook('cast', {by: 'player'}, this.onCast)
+
+		this.addHook('applybuff',
+			{
+				by: 'player',
+				abilityId: RELEVANT_STATUSES,
+			},
+			() => this.buffs++)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private onCast(event: CastEvent) {
+		const actionId = event.ability.guid
+
+		// cases to ignore, No Mercy any other modules will handle the latter 2.
+
+		if (actionId === ACTIONS.ATTACK.id || actionId === ACTIONS.BURST_STRIKE.id || actionId === ACTIONS.SONIC_BREAK.id ) {
+			return
+		}
+
+		if (actionId === ACTIONS.GNASHING_FANG.id) {
+
+			this.onEndGnashingCombo(event) // Make damn sure the previous window has ended.
+
+			const gnashingComboState = new GnashingComboState(event.timestamp)
+			const fightTimeRemaining = this.parser.fight.end_time - event.timestamp
+			// gnashingComboState.isRushing = GNASHING_TIME >= fightTimeRemaining
+
+			this.gnashingComboWindows.push(gnashingComboState)
+		}
+
+		if (COMBO_BREAKERS.hasOwnProperty(actionId) ) {
+			this.onEndGnashingCombo(event)
+		}
+
+		// If the action is a gnashingCombo one, log it
+
+		const lastGnashingCombo = this.lastGnashingCombo
+		if (lastGnashingCombo != null && lastGnashingCombo.end == null) {
+
+				switch (actionId) {
+					case ACTIONS.GNASHING_FANG.id:
```

Comment:
Couldn't this be heavily simplified by just 
```
if (COMBO_ACTIONS.contains(actionId))
    lastGnashingCombo.rotation.push(event)
```
and then using lastGnashingCombo.rotation.length as comparison for the isProper check?

Especially if you go with just dropping tracking the window end time altogether, could streamline the logic pretty heavily.  You could also in that case make isProper and the various numAction properties be computed class getters (see NormalisedEvents in core) and not have to track any of that.

Reply:
It could. I'll get on that.